src/services/item/utils.ts

```diff
@@ -133,3 +139,95 @@ export const getPostItemPayloadFromFormData = (
     thumbnail: formData.file,
   };
 };
+
+/**
+ * Apply item search criteria on given select query
+ * @param query current select query builder
+ * @param account member requesting the data
+ * @param ItemSearchParams search parameters
+ * @param entityNames string name of the table defined in select query builder
+ */
+export const applySearchParamsToQuery = (
+  query: SelectQueryBuilder<ObjectLiteral>,
+  account: Account,
+  {
+    creatorId,
+    keywords,
+    sortBy = SortBy.ItemUpdatedAt,
+    ordering = Ordering.DESC,
+    permissions,
+    types,
+  }: ItemSearchParams,
+  { item = 'item', creator = 'creator' }: { item?: string; creator?: string } = {},
+): void => {
+  if (creatorId) {
+    query.andWhere('item.creator = :creatorId', { creatorId });
+  }
+
+  if (permissions) {
+    query.andWhere('im.permission IN (:...permissions)', { permissions });
+  }
+
+  const allKeywords = keywords?.filter((s) => s && s.length);
+  if (allKeywords?.length) {
+    const keywordsString = allKeywords.join(' ');
+    query.andWhere(
+      new Brackets((q) => {
+        // search in english by default
+        q.where(`${item}.search_document @@ plainto_tsquery('english', :keywords)`, {
+          keywords: keywordsString,
+        });
+
+        // no dictionary
+        q.orWhere(`${item}.search_document @@ plainto_tsquery('simple', :keywords)`, {
+          keywords: keywordsString,
+        });
+
+        // raw words search
+        allKeywords.forEach((k, idx) => {
+          q.orWhere(`${item}.name ILIKE :k_${idx}`, {
+            [`k_${idx}`]: `%${k}%`,
+          });
+        });
+
+        // search by member lang
+        const memberLang = isMember(account) ? account.lang : DEFAULT_LANG;
+        if (memberLang != DEFAULT_LANG && ALLOWED_SEARCH_LANGS[memberLang]) {
+          q.orWhere(`${item}.search_document @@ plainto_tsquery(:lang, :keywords)`, {
+            keywords: keywordsString,
+            lang: ALLOWED_SEARCH_LANGS[memberLang],
+          });
+        }
+      }),
+    );
+  }
+
+  if (types) {
+    query.andWhere(`${item}.type IN (:...types)`, { types });
+  }
+
+  if (sortBy) {
+    // map strings to correct sort by column
+    let mappedSortBy;
+    switch (sortBy) {
+      case SortBy.ItemType:
+        mappedSortBy = `${item}.type`;
+        break;
+      case SortBy.ItemUpdatedAt:
+        mappedSortBy = `${item}.updated_at`;
+        break;
+      case SortBy.ItemCreatedAt:
+        mappedSortBy = `${item}.created_at`;
+        break;
+      case SortBy.ItemCreatorName:
+        mappedSortBy = `${creator}.name`;
+        break;
+      case SortBy.ItemName:
+        mappedSortBy = `${item}.name`;
+        break;
+    }
+    if (mappedSortBy) {
+      query.orderBy(mappedSortBy, orderingToUpperCase(ordering));
+    }
+  }
+};
```

Comment:
What do you think of this function? @spaenleh @Defferrard @MatveyK @ReidyT  I used already twice this process (accessible, children) and will have to use it for recycled items too.

Reply:
Doesn't this function do a bit too much ??

Reply:
I can cut in pieces, but then call all of them every time? Have a wrapped function?